cub/device/dispatch/dispatch_radix_sort.cuh

```diff
@@ -1747,7 +1747,10 @@ struct DispatchRadixSort :
         }
 
         // Check if simple copy suffices (is_overwrite_okay == false at this point)
-        if (begin_bit == end_bit)
+        cudaError_t error = cudaSuccess;
+        bool has_uva = false;
+        if ((error = HasUVA(has_uva)) != cudaSuccess) return error;
+        if (begin_bit == end_bit & has_uva)
```

Comment:
This check is not synchronized with the temp storage size estimation above. On systems without uva it'll lead to invalid memory accesses. 

Also, property queries are not free. Can we invoke `HasUVA` only when `begin_bit == end_bit`?

Reply:
Done (for invoking `HasUVA()` only when `begin_bit == end_bit`).

Regarding temporary storage estimate, the one above only applies to the cases where no GPU work at all is performed. If a single copy is required, a separate estimate is provided by `InvokeCopy()` (see the function definition for the actual code).